gcloud-java-dns/src/test/java/com/google/gcloud/dns/testing/LocalDnsHelperTest.java

```diff
@@ -0,0 +1,1405 @@
+  private static final Change CHANGE_COMPLEX = new Change();
+  private static final LocalDnsHelper LOCAL_DNS_HELPER = LocalDnsHelper.create(0L); // synchronous
+  private static final Map<DnsRpc.Option, ?> EMPTY_RPC_OPTIONS = ImmutableMap.of();
+  private static final DnsRpc RPC =
+      new DefaultDnsRpc(LOCAL_DNS_HELPER.options());
+  private static final String REAL_PROJECT_ID = LOCAL_DNS_HELPER.options().projectId();
+  private Map<String, Object> optionsMap;
+
+  @BeforeClass
+  public static void before() {
+    ZONE1.setName(ZONE_NAME1);
+    ZONE1.setDescription("");
+    ZONE1.setDnsName(DNS_NAME);
+    ZONE1.setNameServerSet("somenameserveset");
+    ZONE2.setName(ZONE_NAME2);
+    ZONE2.setDescription("");
+    ZONE2.setDnsName(DNS_NAME);
+    ZONE2.setNameServerSet("somenameserveset");
+    RRSET1.setName(DNS_NAME);
+    RRSET1.setType(RRSET_TYPE);
+    RRSET1.setRrdatas(ImmutableList.of("1.1.1.1"));
+    RRSET2.setName(DNS_NAME);
+    RRSET2.setType(RRSET_TYPE);
+    RRSET2.setRrdatas(ImmutableList.of("123.132.153.156"));
+    RRSET_KEEP.setName(DNS_NAME);
+    RRSET_KEEP.setType("MX");
+    RRSET_KEEP.setRrdatas(ImmutableList.of("255.255.255.254"));
+    CHANGE1.setAdditions(ImmutableList.of(RRSET1, RRSET2));
+    CHANGE2.setDeletions(ImmutableList.of(RRSET2));
+    CHANGE_KEEP.setAdditions(ImmutableList.of(RRSET_KEEP));
+    CHANGE_COMPLEX.setAdditions(ImmutableList.of(RRSET_KEEP));
+    CHANGE_COMPLEX.setDeletions(ImmutableList.of(RRSET_KEEP));
+    LOCAL_DNS_HELPER.start();
+  }
+
+  @Before
+  public void setUp() {
+    resetProjects();
+    optionsMap = new HashMap<>();
+  }
+
+  private static void resetProjects() {
+    for (String project : LOCAL_DNS_HELPER.projects().keySet()) {
+      LOCAL_DNS_HELPER.projects().remove(project);
+    }
+  }
+
+  @AfterClass
+  public static void after() {
+    LOCAL_DNS_HELPER.stop();
+  }
+
+  @Test
+  public void testCreateZone() {
+    ManagedZone created = RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    // check that default records were created
+    DnsRpc.ListResult<ResourceRecordSet> resourceRecordSetListResult
+        = RPC.listDnsRecords(ZONE1.getName(), EMPTY_RPC_OPTIONS);
+    assertEquals(2, Lists.newLinkedList(resourceRecordSetListResult.results()).size());
+    assertEquals(created, LOCAL_DNS_HELPER.findZone(REAL_PROJECT_ID, ZONE1.getName()).zone());
+    ManagedZone zone = RPC.getZone(ZONE_NAME1, EMPTY_RPC_OPTIONS);
+    assertEquals(created, zone);
+    try {
+      RPC.create(null, EMPTY_RPC_OPTIONS);
+      fail("Zone cannot be null");
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(400, ex.code());
+      assertTrue(ex.getMessage().contains("entity.managedZone"));
+    }
+    // create zone twice
+    try {
+      RPC.create(ZONE1, EMPTY_RPC_OPTIONS);
+    } catch (DnsException ex) {
+      // expected
+      assertEquals(409, ex.code());
+      assertTrue(ex.getMessage().contains("already exists"));
+    }
+    // field options
```

Comment:
We first had it implemented without options. @aozarov raised the question of why the options are not supported and I just accepted it is as something I omitted by accident. There was no discussion. However, I see two things to discuss here:

1) Should the create with options functionality be removed from the helper as I believe that we are aiming to provide the user with an independent local implementation that can be used for more than just testing gcloud-java library
2) Should the create with options functionality be removed from the gcloud-java API itself. 

/cc @aozarov